### DISCUSSION TITLE
Can't edit or create .npmrc in monorepo subprojects

### DISCUSSION THREAD

**Discussion by v41e** [aws/aws-pdk#677] [Q&A] [answered]
I don't if it's related to the fix made in https://github.com/aws/aws-pdk/issues/376 but I can't edit or create a `.npmrc` file in a subproject of a monorepo

Example:

```ts
const construct = new projen.awscdk.AwsCdkConstructLibrary({
  ...baseOptions,

  name: 'construct',
  packageName: '@acme/construct',
  parent: monorepo,
  outdir: 'packages/construct',

  cdkVersion: '2.118.0',
})

// Edit npmrc object natively available don't work
construct.npmrc.addConfig('key', 'value')
construct.npmrc.addRegistry('https://example.com/', '@acme')

// Create a new npmrc don't work too
const constructNpmrc = new projen.javascript.NpmConfig(construct)
constructNpmrc.addConfig('key', 'value')
constructNpmrc.addRegistry('https://example.com/', '@acme')
```

For now I can create the file manually, but I don't if i did something wrong of if it's a bug an should be reported

**Comment by v41e** [comment]
Context: I want to publish some of my packages in my nonorepo in gitlab private registry

**Reply by v41e** [reply]
Also how I am supposed to upgrade version in subproject? do i need to make subprojects releasable too in projenrc.ts ?

**Reply by cogwirrel** [reply]
Ah interesting - is it a requirement to have the `.npmrc` defined at the subproject level in order to publish those packages, or can settings be added to the root `.npmrc`?

The fix you referenced is indeed why the `.npmrc` is being removed - if you can't achieve what's needed by editing the root `.npmrc` we might need to consider alternative options!

**Reply by v41e** [reply]
The alternative option is simple, I create .npmrc file manually

I would like to edit root .npmrc file but i don't know how to publish the subprojects from the root, when i run npm publish from the root it said that i cannot publish because package is private and the monorepo is obviously private because it's not supposed to be published.

Any idea of how to publish subpackages from the root? I looked at the PDK release management there is a subpackage pdk that sync all other subpackages into PDK package and then publish to all package registries but that's overcomplicated for my case, a simple publish is enought

And i didn't understood how the PDK version is upgraded because the version is locked at 0.0.0 in package.json

**Reply by agdimech** [reply, marked as answer]
Hi - are you able to just create a publish task in each package you want to publish and then just call `pdk nx run-many -t publish` from the root?
